File: sdks/php/src/Analytics/Requests/GetChurnCohortsAnalyticsRequest.php
<?php

namespace NovaBilling\Analytics\Requests;

use NovaBilling\Core\Json\JsonSerializableType;

class GetChurnCohortsAnalyticsRequest extends JsonSerializableType
{
    /**
     * @var ?float $months Number of months to analyze (default 12)
     */
    public ?float $months;

    /**
     * @param array{
     *   months?: ?float,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->months = $values['months'] ?? null;
    }
}

File: sdks/csharp/src/NovaBilling/Analytics/Requests/GetChurnCohortsAnalyticsRequest.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record GetChurnCohortsAnalyticsRequest
{
    /// <summary>
    /// Number of months to analyze (default 12)
    /// </summary>
    [JsonIgnore]
    public double? Months { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
